libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -312,6 +312,12 @@ void WaveTrack::SetOffset(double o)
    mOffset = o;
 }
 
+void WaveTrack::OnProjectTempoChange(
+   const std::optional<double>& oldTempo, double newTempo)
+{
+   mClipList.OnProjectTempoChange(oldTempo, newTempo);
```

Comment:
The extra I direction of WaveClipList serves no purpose.

Reply:
You know it gets annoying when you're (probably) right for the 10th time in a row?

Reply:
Now I remember why this is needed.

If a clip is cut from project A and pasted in B, it needs to know immediately what the tempo of B is to be painted and played back correctly. The clip copy constructor cannot rely on that information from the other clip.

When do we know for sure when a clip becomes visible in a project? I can only think of this moment, when the clip is pushed into a track's clip list. This class is here to make sure that this doesn't happen without a tempo update.

But it doesn't have to be part of this very PR, we'll have time to debate over this.

Reply:
`WaveClipList` only added a `std::optional<double>` data member, which could just as easily become part of `WaveTrack.`

Reply:
No, if you call `WaveClipList::PushBack`, it sets the project tempo on the clip.

```
void WaveClipList::PushBack(WaveClipHolder clip)
{
   if (mProjectTempo.has_value())
      clip->SetProjectTempo(*mProjectTempo);
   mClips.push_back(std::move(clip));
}
```

Reply:
Still, you might simply change every `push_back` in WaveTrack to use instead a new private member function of WaveTrack that does this.